src/components/updatemenus/scrollbox.js

```diff
@@ -0,0 +1,406 @@
+        w = this.position.w,
+        t = this.position.t,
+        h = this.position.h,
+        boxW = w,
+        boxH = h,
+        boxL, boxR,
+        boxT, boxB;
+
+    if(boxW > fullWidth) boxW = fullWidth / 4;
+    if(boxH > fullHeight) boxH = fullHeight / 4;
+
+    var minSize = 4 + (ScrollBox.barLength + 2 * ScrollBox.barPad);
+    boxW = Math.max(boxW, minSize);
+    boxH = Math.max(boxH, minSize);
+
+    if(0 <= l && l <= fullWidth) {
+        boxL = l;
+        boxR = boxL + boxW;
+    }
+    else {
+        // align left
+        boxL = 0;
+        boxR = boxL + boxW;
+    }
+
+    if(0 <= t && t <= fullHeight) {
+        boxT = t;
+        boxB = boxT + boxH;
+    }
+    else {
+        // align top
+        boxT = 0;
+        boxB = boxT + boxH;
+    }
+
+    if(boxR > fullWidth) {
+        // align right
+        boxR = fullWidth;
+        boxL = boxR - boxW;
+    }
+
+    if(boxB > fullHeight) {
+        // align bottom
+        boxB = fullHeight;
+        boxT = boxB - boxH;
+    }
+
+    this._box = {
+        l: boxL,
+        t: boxT,
+        w: boxW,
+        h: boxH
+    };
+
+    // compute position of horizontal scroll bar
+    var needsHorizontalScrollBar = (w > boxW),
+        hbarW = ScrollBox.barLength + 2 * ScrollBox.barPad,
+        hbarH = ScrollBox.barWidth + 2 * ScrollBox.barPad,
+        hbarL = boxL,
+        hbarT = (boxB + hbarH < fullHeight) ? boxB : fullHeight - hbarH;
+
+    var hbar = this.container.selectAll('rect.scrollbar-horizontal').data(
+            (needsHorizontalScrollBar) ? [0] : []);
+
+    hbar.exit()
+        .on('.drag', null)
+        .remove();
+
+    hbar.enter().append('rect')
+        .classed('scrollbar-horizontal', true)
+        .call(Color.fill, ScrollBox.barColor);
+
+    if(needsHorizontalScrollBar) {
+        this._hbar = hbar.attr({
+            'rx': ScrollBox.barRadius,
+            'ry': ScrollBox.barRadius,
+            'x': hbarL,
+            'y': hbarT,
+            'width': hbarW,
+            'height': hbarH
+        });
+
+        // hbar center moves between hbarXMin and hbarXMin + hbarTranslateMax
+        this._hbarXMin = hbarL + hbarW / 2;
+        this._hbarTranslateMax = boxW - hbarW;
+    }
+    else {
+        delete this._hbar;
+        delete this._hbarXMin;
+        delete this._hbarTranslateMax;
+    }
+
+    // compute position of vertical scroll bar
+    var needsVerticalScrollBar = (h > boxH),
+        vbarW = ScrollBox.barWidth + 2 * ScrollBox.barPad,
+        vbarH = ScrollBox.barLength + 2 * ScrollBox.barPad,
+        vbarL = (boxR + vbarW < fullWidth) ? boxR : fullWidth - vbarW,
+        vbarT = boxT;
+
+    var vbar = this.container.selectAll('rect.scrollbar-vertical').data(
+            (needsVerticalScrollBar) ? [0] : []);
+
+    vbar.exit()
+        .on('.drag', null)
+        .remove();
+
+    vbar.enter().append('rect')
+        .classed('scrollbar-vertical', true)
+        .call(Color.fill, ScrollBox.barColor);
```

Comment:
Would it be ok to chain

```js
.attr('opacity', '0')
.transition()
.attr('opacity', '1')
```

here and for `hbar` to make the scroll bar appear at the same time as the buttons?